### PR TITLE
CI docs: use gir-rustdoc

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
This generates a main page along with a popover to point to latest docs if the user is reading an old/git version of it

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -27,6 +27,8 @@ jobs:
         run: cargo build --release
       - run: cargo install rustdoc-stripper
       - run: python3 ./generator.py --embed-docs --yes ./
+      - run: git clone https://gitlab.gnome.org/World/Rust/gir-rustdoc/ # checkout action doesn't support random urls
+      - run: echo "RUSTDOCFLAGS=$(eval python3 ./gir-rustdoc/gir-rustdoc.py --pages-url 'https://gtk-rs.org/gtk-rs-core/' --default-branch 'master' --branch 'master' pre-docs | xargs)" >> ${GITHUB_ENV}
```
Comment (GuillaumeGomez) [inline .github/workflows/docs.yml:31]: Would have make maybe more sense to put it in the release script? Well, not a blocker in any case.

**Review by GuillaumeGomez — APPROVED** [review]

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -43,3 +45,14 @@ jobs:
...
+      - run: python3 ./gir-rustdoc/gir-rustdoc.py --project-title 'GTK Core Rust bindings' html-index
+      - name: deploy index page
+        uses: peaceiris/actions-gh-pages@v3
+        if: ${{ (github.event_name == 'push' && github.ref == 'refs/heads/master') || github.event_name == 'release' }}
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./public/
+          keep_files: true
+          destination_dir: ./
+
```
Comment (GuillaumeGomez) [inline .github/workflows/docs.yml:58]: Nit: please remove this extra line.

Comment (bilelmoussaoui) [reply .github/workflows/docs.yml]: Done, please squash when merging :P too lazy to clone/rebase...

**Comment by GuillaumeGomez** [comment]
Now please squash. 😛

**Comment by bilelmoussaoui** [comment]
> Now please squash. stuck_out_tongue

you can do that from github's ui :p

**Comment by GuillaumeGomez** [comment]
I saw your comment afterwards, no problem. ;)

**Comment by bilelmoussaoui** [comment]
@GuillaumeGomez both are ready to be merged :) and once deployed we will have nice pages https://gtk-rs.org/gtk3-rs/ & https://gtk-rs.org/gtk-rs-core/ like this one https://gtk-rs.org/gtk4-rs.

**Comment by bilelmoussaoui** [comment]
actually let's wait til lhttps://gitlab.gnome.org/World/Rust/gir-rustdoc/-/merge_requests/8 is merged otherwise it fails due to another change i made

**Review by sdroege — APPROVED** [review]

**Pull request merged**